app/models/api_key.rb

```diff
@@ -39,7 +39,10 @@ def verify!
 
   def self.new_key
     result = String.random(KEY_LENGTH)
-    key = String.random(KEY_LENGTH) while find_by_key(result)
+    # rubocop:disable Lint/UselessAssignment:
+    # RuboCop gives false positive
+    key = String.random(KEY_LENGTH) while find_by(key: result)
+    # rubocop:enable Lint/UselessAssignment:
     result
```

Comment:
I don't completely understand this. (My fault, not yours; I'm just not as fluent in Ruby as the rest of you.)
`key = String.random(KEY_LENGTH) while find_by(key: result)` seems quite elegant (though it took me a while to understand it).
**But is the return value correct?** 
In particular, I don't understand why it returns `result` rather than `key` when find_by(key: result) finds something.

Reply:
I don't think this is correct.  I think the intended code is `result = String.random(KEY_LENGTH) while find_by(key: result)`.  I suspect we've been getting away it because `find_by_key(String.random(KEY_LENGTH)` is extremely unlikely to return true.